### PR TITLE
Specify Platforms in all actions

### PR DESCRIPTION
**Pull request by RubenKid**

### CODE REVIEW

**Comment by RubenKid** [comment]
@espinchi I reverted commit

File: demo/src/main/java/com/pixable/trackingwrap/demo/MainActivity.java [hunk 1]
```diff
@@ -21,8 +21,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        Pixalytics.get().onSessionFinish(this, new String[]{FlurryPlatform.ID});
+    }
+
     public void onFlurryClick(View view) {
-        Pixalytics.get().trackEvent(this, new Event.Builder().name("Foo").build(), FlurryPlatform.ID);
+        Pixalytics.get().trackEvent(this, new Event.Builder().name("Foo").build(), new String[]{FlurryPlatform.ID});
```
Comment (espinchi) [inline demo/src/main/java/com/pixable/trackingwrap/demo/MainActivity.java]: There's no need to turn `FlurryPlatform.ID` into `new String[]{FlurryPlatform.ID}`. By using varargs for that parameter, we can have a more natural syntax

File: pixalytics-core/src/main/java/com/pixable/pixalytics/core/Pixalytics.java [hunk 2]
```diff
@@ -208,6 +216,12 @@ public void trackScreen(Context context, Screen screen) {
         }
     }
 
+    private Set<Platform> checkAndGetPlatformsFromIds(String[] platformIds) {
```
Comment (espinchi) [inline pixalytics-core/src/main/java/com/pixable/pixalytics/core/Pixalytics.java]: I'm trying to overlook this method, but I can't help asking: are you sure `platforms = checkAndGetPlatformsFromIds()` is better than having

```
checkAppIsInitialized();
checkPlatformsAreValid(platformIds);

platforms = idsToPlatforms(platformIds);
```

?

Even if it's slightly shorter, don't you think it makes the methods harder to read? Also, it mixes methods that are pretty different.

I guess I can be convinced that this is better, but I'm gonna need some arguments :)

File: demo/src/main/java/com/pixable/trackingwrap/demo/MainActivity.java (hunk 1 above)
Comment (RubenKid) [inline demo/src/main/java/com/pixable/trackingwrap/demo/MainActivity.java]: you are completely right

File: pixalytics-core/src/main/java/com/pixable/pixalytics/core/Pixalytics.java (hunk 2 above)
Comment (RubenKid) [inline pixalytics-core/src/main/java/com/pixable/pixalytics/core/Pixalytics.java]: The main reason as you said was only to simplify the code. I think is good to have just one line to make all checks that could stop us of making the action. Is true that as you said we are making a check and search in same function, so...for simplicity I would go for unify both check functions in one and leave the "getPlatformsFromIds" as a separate one

**Comment by RubenKid** [comment]
Good!

**Pull request merged**